src/test/osd/test_scrubber_be.cc

```diff
@@ -0,0 +1,870 @@
+    const pg_pool_t* ext_pool_info = osdmap->get_pg_pool(pool_id);
+    pool =
+      std::make_shared<PGPool>(osdmap, pool_id, *ext_pool_info, pool_conf.name);
+
+    std::cout << "pool: " << pool->info << std::endl;
+
+    // a PG in that pool?
+    info = setup_pg_in_map();
+    // std::cout << fmt::format("info: {}\n", info);
+    std::cout << "info: " << info << std::endl;
+
+
+    real_objs_list =
+      ScrubGenerator::make_real_objs_conf(pool_id, real_objs, acting_osds);
+
+    // now we can create the main mockers
+
+    // the "PgScrubber"
+    test_scrubber = std::make_unique<TestScrubber>(spg, osdmap, &logger);
+    std::cout << "test_scrubber: "
+              << "test scrubber created" /* *test_scrubber*/ << std::endl;
+
+    // the "Pg" (and its backend)
+    test_pg = std::make_unique<TestPg>(pool, info, i_am);
+    std::cout << fmt::format("{}: test_pg: OK\n", __func__);
+    std::cout << fmt::format("{}: acting: {}\n", __func__, acting_shards);
+    sbe = std::make_unique<TestScrubBackend>(*test_scrubber,
+                                             *test_pg,
+                                             i_am,
+                                             /* repair? */ false,
+                                             scrub_level_t::deep,
+                                             acting_shards);
+
+    // create a osd-num only copy of the relevant OSDs
+    acting_osds.reserve(acting_shards.size());
+    for (const auto& shard : acting_shards) {
+      acting_osds.push_back(shard.osd);
+    }
+
+    sbe->new_chunk();
+    std::cout << fmt::format("{}: TestScrubBackend created\n", __func__);
+    fake_a_scrub_set(real_objs_list);
+    std::cout << fmt::format("{}: fake_a_scrub_set() done\n", __func__);
+  }
+
+  void TearDown() override;
+
+  // void fake_a_scrub_set(ScrubGenerator::RealObjsConf conf);
+  void fake_a_scrub_set(ScrubGenerator::RealObjsConfList& all_sets);
+
+  std::unique_ptr<TestScrubBackend> sbe;
+
+  // I am the primary
+  //      pg_shard_t i_am{0, pool_id};
+  spg_t spg;
+
+  pg_shard_t i_am;  // my osd and no shard
+  std::set<pg_shard_t> acting_shards;
+  std::vector<int> acting_osds;
+  int acting_primary;
+
+  std::unique_ptr<TestScrubber> test_scrubber;
+
+  int64_t pool_id;
+  pg_pool_t pool_info;
+
+  // OSDMap osdmap;
+  OSDMapRef osdmap;
+
+  std::shared_ptr<PGPool> pool;
+  pg_info_t info;
+
+
+  std::unique_ptr<TestPg> test_pg;
+
+  // generated sets of "objects" for the active OSDs
+  ScrubGenerator::RealObjsConfList real_objs_list;
+
+ private:
+  // void setup_map(int num_osds, const ScrubGenerator::pool_conf_t& pconf);
+  OSDMapRef setup_map(int num_osds, const ScrubGenerator::pool_conf_t& pconf);
+
+  pg_info_t setup_pg_in_map();
+};
+
+
+// ///////////////////////////////////////////////////////////////////////////
+// ///////////////////////////////////////////////////////////////////////////
+
+
+// copied from TestOSDMap.cc
```

Comment:
Probably worth separating out into a common header.

Reply:
It was more of an inspiration source.. The specific data will be changed later on, when the tests are improved. Didn't see a point in extracting that part.